client/tufclient.go

```diff
@@ -78,25 +79,81 @@ func (c *TUFClient) update() error {
 	return nil
 }
 
-// downloadRoot is responsible for downloading the root.json
-func (c *TUFClient) downloadRoot() error {
+// updateRoot checks if there is a newer version of the root available, and if so
+// downloads all intermediate root files to allow proper key rotation.
+func (c *TUFClient) updateRoot() error {
 	role := data.CanonicalRootRole
-	consistentInfo := c.newBuilder.GetConsistentInfo(role)
 
-	// We can't read an exact size for the root metadata without risking getting stuck in the TUF update cycle
-	// since it's possible that downloading timestamp/snapshot metadata may fail due to a signature mismatch
-	if !consistentInfo.ChecksumKnown() {
-		logrus.Debugf("Loading root with no expected checksum")
+	// Get Current Root Version
+	currentRootConsistentInfo := c.oldBuilder.GetConsistentInfo(role)
+	currentRoot, _ := c.cache.GetSized(role, -1)
+	c.oldBuilder.Load(currentRootConsistentInfo.RoleName, currentRoot, 1, true)
+	currentVersion := c.oldBuilder.GetLoadedVersion(currentRootConsistentInfo.RoleName)
+
+	// Get New Root Version
+	raw, err := c.remote.GetSized(data.CanonicalRootRole, -1)
```

Comment:
`c.tryLoadRemote` and  `c.c.tryLoadCacheThenRemote` both return the failed root bytes, if it fails to load.  If we can make `c.downloadRoot` also do so, then you can probably just call that here, and if it succeeds, we can skip the rest of the logic to check versions, since we have to do `downloadRoot` if it's a version behind anyway. 

If it fails, and it's not due to a `trustpinning.ErrRootRotationFail` (for instance if it's expired, invalid, or otherwise fails trust pinning), we can also bail without downloading intermediate versions.